state/memory.go

```diff
@@ -425,7 +430,8 @@ func (db *inMemoryStateDB) GetLogs(txHash common.Hash, blk uint64, blkHash commo
 	// transaction, all logs are from the same transactions. But
 	// those need to be collected in the right order (inverse order
 	// snapshots).
-	logs := collectLogs(db.state)
+	logs := collectLogs(db.state, blkTimestamp)
```

Comment:
This looks odd ... in `collectLogs` you are filtering for the correct block times but, but the times are only set in the logs in the loop below.